exe/wallet/Main.hs

```diff
@@ -173,20 +182,34 @@ exec manager args
             (ApiT wPwd)
 
     | args `isPresent` command "wallet" && args `isPresent` command "update" = do
-        wId <- args `parseArg` longOption "id"
+        wId <- args `parseArg` longOption "wallet-id"
         wName <- args `parseArg` longOption "name"
         runClient $ putWallet (ApiT wId) $ WalletPutData
             (Just $ ApiT wName)
 
     | args `isPresent` command "wallet" && args `isPresent` command "delete" = do
-        wId <- args `parseArg` longOption "id"
+        wId <- args `parseArg` longOption "wallet-id"
         runClient $ void $ deleteWallet (ApiT wId)
 
+    | args `isPresent` command "transaction" && args `isPresent` command "create" = do
+        wId <- args `parseArg` longOption "wallet-id"
+        ts <- args `parseAllArgs` longOption "target"
+        (wPwd, _) <- do
+            let prompt = "Please enter a passphrase: "
+            let parser = fromText @(Passphrase "encryption")
+            getSensitiveLine prompt Nothing parser
```

Comment:
This applies the password validation rules (e.g. 10 chars). But if the password was set through an API call with curl, then it might not validate.

Reply:
@rvl Not sure to understand what you mean here? What API call with cURL? To the wallet server API ?

Reply:
I believe this is fixed by Matthias

Reply:
The code is now asking for the encryption password twice when submitting a transaction.

Reply:
@rvl I am not sure are you ok with this change?

> This applies the password validation rules (e.g. 10 chars). But if the password was set through an API call with curl, then it might not validate.

password wouldn't validate with curl as well because validation rules are enforced in core (have a look at `FromText` and `FromJSON`). So this 10 char password isn't enforced only on cardano-wallet CLI client (as I think you might thought)

Reply:
There are two things here:

1/ We do indeed perform validation at the API-level too. Whether we should allow empty passwords from the CLI or the API is another question, out-of-scope for this PR.

2/ Rodney's second remark is about the small refactoring you did, moving `getPassphrase` into the `where` section. This is actually wrong for sending transaction, as we would prompt the passphase twice in that case! We only need to prompt it once when creating transaction. This needs fix.

Reply:
2) Ah true - didn't understand what Rodney meant to say

Reply:
I did not realise the API enforces a 10 char minimum encryption password (weird idea), therefore I thought the CLI was incorrectly using different validation rules to the API.

Reply:
That's what I thought! We completely share the same decoders in the CLI and API, so any rules applied to decoding values in the API should apply in the CLI. 
In the future, we would even enforce some deeper validation rules for the passphrase (comparing against a list of mostly used and enforcing some good entropy in the passphrase) because that's the first step towards securing your wallet as a user. 

Truth is, if we don't do that, people put an empty mnemonic and an empty passphrase and then, ask the support and their lawyers why their funds disappeared from their wallet :disappointed: